tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -394,5 +450,19 @@ public void TestConvertObjectIdToObjectId()
 
             Assert.Equal(oid, oidConverted);
         }
+
+        // nested types
+        private class AppDomainCourier : MarshalByRefObject
```

Comment:
After thinking hard about what this class is doing I suggest renaming to `ObjectIdCreator`.

Reply:
The idea of this class that this class should delivery `ObjectId` to a different AppDomain. That's why `Courier`.
But I don't mind about your naming. Done